src/Microsoft.AspNet.Authentication.Facebook/FacebookAppBuilderExtensions.cs

```diff
@@ -44,13 +23,15 @@ public static IApplicationBuilder UseFacebookAuthentication(this IApplicationBui
             {
                 throw new ArgumentNullException(nameof(app));
             }
-
-            var options = new FacebookOptions();
-            if (configureOptions != null)
+            if (configureOptions == null)
             {
-                configureOptions(options);
+                throw new ArgumentNullException(nameof(configureOptions));
```

Comment:
I've made `configureOptions` explicitly required but should we allow null configure options? I've seen both patterns being used. Some places even default `configureOptions` to null so that it's an optional parameter.